src/Tests/Tests/Test_Vibrato.cpp

```diff
@@ -3,19 +3,331 @@
 #ifdef WITH_TESTS
 #include "Vector.h"
 #include "Vibrato.h"
-
+#include "RingBuffer.h"
 #include "gtest/gtest.h"
-
+#include "Synthesis.h"
 
 namespace vibrato_test {
     void CHECK_ARRAY_CLOSE(float* buffer1, float* buffer2, int iLength, float fTolerance)
     {
         for (int i = 0; i < iLength; i++)
         {
+            float error = abs(buffer1[i] - buffer2[i]);
             EXPECT_NEAR(buffer1[i], buffer2[i], fTolerance);
         }
     }
 
+    class RingBuffer : public testing::Test
+    {
+    protected:
+        void SetUp() override
+        {
+            pCRingBuff = new CRingBuffer<float>(kBlockSize);
+        }
+
+        virtual void TearDown()
+        {
+            delete pCRingBuff;
+            pCRingBuff = 0;
+        }
+
+        CRingBuffer<float>* pCRingBuff = 0;
+        static const int kBlockSize = 17;
+    };
+
+    class Vibrato : public testing::Test
+    {
+    protected:
+        void SetUp() override
+        {
+            CVibrato::create(pCVibrato);
+        }
+
+        virtual void TearDown()
+        {
+            CVibrato::destroy(pCVibrato);
+        }
+
+        float* pfInputBuffer;
+        float* pfOutputBuffer;
+        CVibrato* pCVibrato;
+    };
+
+    TEST_F(RingBuffer, BufferLength)
+    {
+        EXPECT_EQ(pCRingBuff->getLength(), 17);
+    }
+
+    TEST_F(RingBuffer, IntegerDelay)
+    {
+        for (int i = 0; i < 5; i++)
+        {
+            pCRingBuff->putPostInc(1.F * i);
+        }
+
+        for (int i = 5; i < 30; i++)
+        {
+            EXPECT_EQ(pCRingBuff->getNumValuesInBuffer(), 5);
+            EXPECT_NEAR(pCRingBuff->getPostInc(), i - 5, 1e-3);
```

Comment:
Would you explain the reason why you used NEAR instead of EQ here and in the following Ring Buffer tests?

Reply:
For floating point numbers using equal makes no sense